warehouse/tuf/targets.py

```diff
@@ -0,0 +1,91 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+
+from pyramid.httpexceptions import HTTPBadGateway
+
+from warehouse.packaging.models import File
+from warehouse.packaging.utils import current_simple_details_path, render_simple_detail
+
+
+def _payload(targets):
+    """Helper to create payload for POST or DELETE targets request."""
+    return {
+        "targets": targets,
+        "publish_targets": True,
+    }
+
+
+def _payload_targets_part(path, size, digest):
+    """Helper to create payload part for POST targets request."""
+    return {
+        "path": path,
+        "info": {
+            "length": size,
+            "hashes": {"blake2b-256": digest},
+        },
+    }
+
+
+def _handle(response):
+    """Helper to handle http response for POST or DELETE targets request."""
+    if response.status_code != 202:
+        raise HTTPBadGateway(f"Unexpected TUF Server response: {response.text}")
+
+    return response.json()
```

Comment:
RSTUF, by design, is asynchronous.
Every request to add or remove an artifact in TUF metadata generates a task id.
With this task id is possible to do call backs to receive the status. RSTUF gives the control to monitor this task to the requester.

Some integration design questions about Warehouse
* Should the Warehouse create an asynchronous monitoring/action for the tasks to take some action depending on the result?

* Should Warehouse store the task result in some Warehouse table as in RSTUF use the default [Celery result expires]?  Currently, we store the task id in the events, but the task result is not persistent as mentioned, for example: https://github.com/pypi/warehouse/blob/818210ad8647bad2f29f4494fc4a7cf13d63995b/warehouse/forklift/legacy.py#L1410-L1427